model/task_lifecycle_test.go

```diff
@@ -1279,6 +1282,84 @@ func TestUpdateBuildStatusForTask(t *testing.T) {
 
 }
 
+func TestUpdateVersionAndPatchStatusForBuilds(t *testing.T) {
+	require.NoError(t, db.ClearCollections(build.Collection, patch.Collection, task.Collection, VersionCollection))
+	uiConfig := evergreen.UIConfig{Url: "http://localhost"}
+	require.NoError(t, uiConfig.Set())
+	sender := send.MakeInternalLogger()
+	assert.NoError(t, grip.SetSender(sender))
+	b := &build.Build{
+		Id:        "buildtest",
+		Status:    evergreen.BuildFailed,
+		Requester: evergreen.GithubPRRequester,
+		Version:   "aaaaaaaaaaff001122334455",
+		Activated: true,
+	}
+	p := &patch.Patch{
+		Id:              patch.NewId(b.Version),
+		Status:          evergreen.PatchFailed,
+		GithubPatchData: thirdparty.GithubPatch{HeadOwner: "q"},
+	}
+	v := &Version{
+		Id:        b.Version,
+		Status:    evergreen.VersionFailed,
+		Requester: evergreen.GithubPRRequester,
+	}
+	testTask := task.Task{
+		Id:        "testone",
+		Activated: true,
+		BuildId:   b.Id,
+		Project:   "sample",
+		Status:    evergreen.TaskUndispatched,
+		Version:   b.Version,
+	}
+	anotherTask := task.Task{
+		Id:        "two",
+		Activated: true,
+		BuildId:   b.Id,
+		Project:   "sample",
+		Status:    evergreen.TaskFailed,
+		StartTime: time.Now().Add(-time.Hour),
+		Version:   b.Version,
+	}
+
+	assert.NoError(t, b.Insert())
+	assert.NoError(t, p.Insert())
+	assert.NoError(t, v.Insert())
+	assert.NoError(t, testTask.Insert())
+	assert.NoError(t, anotherTask.Insert())
+
+	assert.NoError(t, UpdateVersionAndPatchStatusForBuilds([]string{b.Id}))
+	// Confirm that the build status was updated to pending in the PR by checking for appropriate Splunk logs
```

Comment:
We don't need to specify that this is Splunk since we could change out logging backend at any point. I think it's sufficient to say something like "confirm if there was a log indicating the build status updated to pending".